src/Forms/Shared/Samples/Location/IndoorPositioning/IndoorPositioning.xaml.cs

```diff
@@ -0,0 +1,204 @@
+            {
+                try
+                {
+                    // WARNING: Never hardcode login information in a production application. This is done solely for the sake of the sample.
+                    return await AuthenticationManager.Current.GenerateCredentialAsync(info.ServiceUri, sampleUser, samplePass, info.GenerateTokenOptions);
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex.Message);
+                    return null;
+                }
+            });
+
+            try
+            {
+                try
+                {
+                    Xamarin.Essentials.PermissionStatus status = await Xamarin.Essentials.Permissions.RequestAsync<Xamarin.Essentials.Permissions.LocationWhenInUse>();
+                    if (status != Xamarin.Essentials.PermissionStatus.Granted)
+                    {
+                        await Application.Current.MainPage.DisplayAlert("Error", "Location permission required for use of indoor positioning.", "OK");
+                        return;
+                    }
+                }
+                catch
+                {
+                    return;
+                }
+
+#if __ANDROID__
+                // Get bluetooth permission for Android devices. Devies running Android 12 or higher need the `BluetoothScan` permission. Android 11 and below require the `Bluetooth` and `BluetoothAdmin` permissions.
+                bool bluetoothScanGranted = await MainActivity.Instance.AskForBluetoothPermission();
+                if (!bluetoothScanGranted)
+                {
+                    await Application.Current.MainPage.DisplayAlert("Error", "Bluetooth permission is required for use of indoor positioning.", "OK");
+                    return;
+                }
+#endif
+
+                // Create a portal item for the web map.
+                ArcGISPortal portal = await ArcGISPortal.CreateAsync(_portalUri, true);
+                PortalItem item = await PortalItem.CreateAsync(portal, ItemId);
+
+                // Load the map in the map view.
+                MyMapView.Map = new Map(item);
+                await MyMapView.Map.LoadAsync();
+
+                // Get the positioning table from the map.
+                await Task.WhenAll(MyMapView.Map.Tables.Select(table => table.LoadAsync()));
+                FeatureTable positioningTable = MyMapView.Map.Tables.FirstOrDefault(table => table.TableName.Equals(PositioningTableName));
+                if (positioningTable == null) return;
+
+                // Get a table of all of the indoor pathways.
+                FeatureLayer pathwaysFeatureLayer = MyMapView.Map.OperationalLayers.FirstOrDefault(l => l.Name.Equals(PathwaysLayerName, StringComparison.InvariantCultureIgnoreCase)) as FeatureLayer;
+                ArcGISFeatureTable pathwaysTable = pathwaysFeatureLayer.FeatureTable as ArcGISFeatureTable;
```

Comment:
`pathwaysFeatureLayer` could be null here, but there's no null-check. If you don't want to risk getting a null, use `Single()` instead of `FirstOrDefault()` in the line above (it'll then throw if it doesn't find the layer, and only one layer)